sql/core/src/main/scala/org/apache/spark/sql/execution/Window.scala

```diff
@@ -498,7 +548,13 @@ private[execution] final class SlidingWindowFunctionFrame(
     ubound: BoundOrdering) extends WindowFunctionFrame {
 
   /** Rows of the partition currently being processed. */
-  private[this] var input: ArrayBuffer[InternalRow] = null
+  private[this] var input: ExternalRowBuffer = null
+
+  /** The next row from `input`. */
+  private[this] var nextRow: InternalRow = null
+
+  /** The rows within current sliding window. */
+  private[this] val buffer: util.Queue[InternalRow] = new util.ArrayDeque[InternalRow]()
```

Comment:
Nit/Performance Paranoia: I would not type it as a ```Queue``` but as an ```ArrayDeque``` (or have scala infer the type). In order to ever prevent megamorphic calls.